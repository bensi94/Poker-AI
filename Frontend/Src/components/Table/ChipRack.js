import React from 'react';
import PropTypes from 'prop-types';

import BlackChip from '../../../assets/black-chip-vertical.svg';
import BlueChip from '../../../assets/blue-chip-vertical.svg';
import GreenChip from '../../../assets/green-chip-vertical.svg';
import PinkChip from '../../../assets/pink-chip-vertical.svg';
import RedChip from '../../../assets/red-chip-vertical.svg';

class ChipRack extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            maxChipCount: 19,
            left: 38.8,
            leftAdd: 3.28,
            topConstant: 4,
            topPercent: 0.6,
            widthConstant: 0.027,
            componentList: [
                'GreenChip',
                'RedChip',
                'BlueChip',
                'PinkChip',
                'BlackChip',
                'BlueChip',
                'RedChip'
            ],
            chipRack: undefined
        };
    }

    componentDidMount() {
        this.setState({
            chipRack: this.getChipRack()
        });
    }

    getChipComponent(component, style, keyString) {
        switch (component) {
            case 'GreenChip':
                return (<GreenChip
                    style={style}
                    className="vertical-chip"
                    width={this.props.tableSize * this.state.widthConstant}
                    key={keyString} />
                );
            case 'RedChip':
                return (<RedChip
                    style={style}
                    className="vertical-chip"
                    width={this.props.tableSize * this.state.widthConstant}
                    key={keyString} />
                );
            case 'BlueChip':
                return (<BlueChip
                    style={style}
                    className="vertical-chip"
                    width={this.props.tableSize * this.state.widthConstant}
                    key={keyString} />
                );
            case 'PinkChip':
                return (<PinkChip
                    style={style}
                    className="vertical-chip"
                    width={this.props.tableSize * this.state.widthConstant}
                    key={keyString} />
                );
            case 'BlackChip':
                return (<BlackChip
                    style={style}
                    className="vertical-chip"
                    width={this.props.tableSize * this.state.widthConstant}
                    key={keyString}/>
                );
            default:
                break;
        }
    }

    getChipRack() {
        let chips = [];
        this.state.componentList.map((component, index) => {
            const randomNumb = Math.floor((Math.random() * this.state.maxChipCount) + 1);
            for (let i = 0; i < randomNumb; i++) {
                const style = {
                    left: `${this.state.left + this.state.leftAdd * index}%`,
                    top: `${this.state.topConstant + i * this.state.topPercent}%`,
                    zIndex: `${i}`
                };
                chips.push(this.getChipComponent(component,
                    style, `${index.toString()}_${i.toString()}`));
            }
        });

        return chips;
    }

    render() {
        return (
            <>
                {this.state.chipRack}
            </>
        );
    }
}

ChipRack.propTypes = {
    tableSize: PropTypes.number.isRequired
};

export default ChipRack;
