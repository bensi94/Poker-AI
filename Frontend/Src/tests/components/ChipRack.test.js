import React from 'react';
import { shallow } from 'enzyme';

import ChipRack from '../../components/Table/ChipRack';

describe('ChipRack Test suite', () => {
    let wrapper;

    // This is done because we use Math.random in the component
    // But we can not do a snapshot test with random variables
    global.Math.random = () => 0.5;

    beforeEach(() => {
        wrapper = shallow(<ChipRack tableSize={500} />);
    });

    it('Should render ChipRack(snapshot) correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // Tests that the getChipComponent function works as expected
    it('Should contain the correct style and key on getChipComponent', () => {
        const styleObj = {
            left: '10%',
            top: '20%',
            zIndexL: 1
        };

        wrapper.instance().setState({
            widthConstant: 1
        });

        const responseComp = wrapper.instance().getChipComponent(
            'GreenChip',
            styleObj,
            'test_key'
        );

        expect(responseComp.key).toBe('test_key');
        expect(responseComp.props.style).toBe(styleObj);
        // Based on the tableSize prop and the widthConstant
        expect(responseComp.props.width).toBe(500);
    });

    it('Should not render a chip on invalid componentString', () => {
        const responseComp = wrapper.instance().getChipComponent(
            'INVALIDSTRING'
        );

        expect(responseComp).toBe(undefined);
    });


    it('Should render the correct amount of components in getChipRack', () => {
        wrapper.instance().setState({
            maxChipCount: 1,
            componentList: [
                'GreenChip',
                'RedChip',
                'BlueChip',
                'PinkChip',
                'BlackChip'
            ]
        });

        expect(wrapper.instance().getChipRack().length).toBe(5);
    });

    it('Should be calculating the style(postions) correctly in getChipRack', () => {
        wrapper.instance().setState({
            maxChipCount: 2,
            left: 1,
            leftAdd: 1,
            topConstant: 1,
            topPercent: 1,
            widthConstant: 1,
            componentList: [
                'GreenChip',
                'RedChip'
            ]
        });

        const responseList = wrapper.instance().getChipRack();
        const responseOne = responseList[0];
        const responseTwo = responseList[1];
        const responseThree = responseList[2];

        const styleOne = {
            left: '1%',
            top: '1%',
            zIndex: '0'
        };

        const styleTwo = {
            left: '1%',
            top: '2%',
            zIndex: '1'
        };

        const styleThree = {
            left: '2%',
            top: '1%',
            zIndex: '0'
        };

        // First component
        expect(responseOne.props.width).toBe(500);
        expect(responseOne.props.style).toStrictEqual(styleOne);

        // Second component
        expect(responseTwo.props.width).toBe(500);
        expect(responseTwo.props.style).toStrictEqual(styleTwo);

        // Third component
        expect(responseThree.props.width).toBe(500);
        expect(responseThree.props.style).toStrictEqual(styleThree);
    });
});
