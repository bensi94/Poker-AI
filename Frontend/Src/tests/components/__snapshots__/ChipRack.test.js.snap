// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipRack Test suite Should render ChipRack(snapshot) correctly 1`] = `
<Fragment>
  <stubbed-asset
    className="vertical-chip"
    key="0_0"
    style={
      Object {
        "left": "38.8%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_1"
    style={
      Object {
        "left": "38.8%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_2"
    style={
      Object {
        "left": "38.8%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_3"
    style={
      Object {
        "left": "38.8%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_4"
    style={
      Object {
        "left": "38.8%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_5"
    style={
      Object {
        "left": "38.8%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_6"
    style={
      Object {
        "left": "38.8%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_7"
    style={
      Object {
        "left": "38.8%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_8"
    style={
      Object {
        "left": "38.8%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="0_9"
    style={
      Object {
        "left": "38.8%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_0"
    style={
      Object {
        "left": "42.08%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_1"
    style={
      Object {
        "left": "42.08%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_2"
    style={
      Object {
        "left": "42.08%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_3"
    style={
      Object {
        "left": "42.08%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_4"
    style={
      Object {
        "left": "42.08%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_5"
    style={
      Object {
        "left": "42.08%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_6"
    style={
      Object {
        "left": "42.08%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_7"
    style={
      Object {
        "left": "42.08%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_8"
    style={
      Object {
        "left": "42.08%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="1_9"
    style={
      Object {
        "left": "42.08%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_0"
    style={
      Object {
        "left": "45.36%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_1"
    style={
      Object {
        "left": "45.36%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_2"
    style={
      Object {
        "left": "45.36%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_3"
    style={
      Object {
        "left": "45.36%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_4"
    style={
      Object {
        "left": "45.36%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_5"
    style={
      Object {
        "left": "45.36%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_6"
    style={
      Object {
        "left": "45.36%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_7"
    style={
      Object {
        "left": "45.36%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_8"
    style={
      Object {
        "left": "45.36%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="2_9"
    style={
      Object {
        "left": "45.36%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_0"
    style={
      Object {
        "left": "48.64%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_1"
    style={
      Object {
        "left": "48.64%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_2"
    style={
      Object {
        "left": "48.64%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_3"
    style={
      Object {
        "left": "48.64%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_4"
    style={
      Object {
        "left": "48.64%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_5"
    style={
      Object {
        "left": "48.64%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_6"
    style={
      Object {
        "left": "48.64%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_7"
    style={
      Object {
        "left": "48.64%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_8"
    style={
      Object {
        "left": "48.64%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="3_9"
    style={
      Object {
        "left": "48.64%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_0"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_1"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_2"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_3"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_4"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_5"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_6"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_7"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_8"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="4_9"
    style={
      Object {
        "left": "51.919999999999995%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_0"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_1"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_2"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_3"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_4"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_5"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_6"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_7"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_8"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="5_9"
    style={
      Object {
        "left": "55.199999999999996%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_0"
    style={
      Object {
        "left": "58.48%",
        "top": "4%",
        "zIndex": "0",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_1"
    style={
      Object {
        "left": "58.48%",
        "top": "4.6%",
        "zIndex": "1",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_2"
    style={
      Object {
        "left": "58.48%",
        "top": "5.2%",
        "zIndex": "2",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_3"
    style={
      Object {
        "left": "58.48%",
        "top": "5.8%",
        "zIndex": "3",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_4"
    style={
      Object {
        "left": "58.48%",
        "top": "6.4%",
        "zIndex": "4",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_5"
    style={
      Object {
        "left": "58.48%",
        "top": "7%",
        "zIndex": "5",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_6"
    style={
      Object {
        "left": "58.48%",
        "top": "7.6%",
        "zIndex": "6",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_7"
    style={
      Object {
        "left": "58.48%",
        "top": "8.2%",
        "zIndex": "7",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_8"
    style={
      Object {
        "left": "58.48%",
        "top": "8.8%",
        "zIndex": "8",
      }
    }
    width={13.5}
  />
  <stubbed-asset
    className="vertical-chip"
    key="6_9"
    style={
      Object {
        "left": "58.48%",
        "top": "9.399999999999999%",
        "zIndex": "9",
      }
    }
    width={13.5}
  />
</Fragment>
`;
